### PR TITLE
On mobile, hide title if logo exists

### PR DESCRIPTION
**Pull request by amyrlam**
## Motivation

Resolves: https://github.com/facebook/Docusaurus/issues/613

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

IS with this PR:

Mobile:
image

Tablet (no change):
image

Desktop (no change):
image

***

WAS:

Mobile:
image

Tablet (no change):
image

Desktop (no change):
image

## Related PRs

It didn't seem like a docs update was necessary?

cc @yangshun @JoelMarcey

### CODE REVIEW

File: lib/static/css/main.css [hunk 1]
```diff
@@ -523,13 +523,12 @@ header h2 {
...
 @media (max-width: 375px) {
-  .logo {
-    display: none;
-  }
-
   .headerTitle {
     font-size: 17px;
   }
+  .headerTitleHidden {
+    display: none !important;
```
Comment (yangshun) [inline lib/static/css/main.css]: Is `!important` needed?

File: lib/core/nav/HeaderNav.js [hunk 2]
```diff
@@ -218,6 +218,12 @@ class HeaderNav extends React.Component {
       (env.translation.enabled
         ? this.props.language + '/versions.html'
         : 'versions.html');
+    let headerClass;
+    if (siteConfig.headerIcon) {
+      headerClass="headerTitleHidden"
```
Comment (yangshun) [inline lib/core/nav/HeaderNav.js]: This is not standard JS style. Should use semicolons, single quotes and space around operators.

Comment (yangshun) [reply lib/core/nav/HeaderNav.js]: Consider simplifying to:

```
const headerClass = siteConfig.headerIcon ? 'headerTitleHidden' : 'headerTitle';
```

File: lib/static/css/main.css (hunk 1 above)
Comment (yangshun) [reply lib/static/css/main.css]: `headerTitleHidden` is not a good name because it's only hidden on smaller widths. How about `headerTitleWithLogo`?

Comment (amyrlam) [reply lib/static/css/main.css]: Yes, `!important` is needed here.

Renamed to `headerTitleWithLogo`.

File: lib/core/nav/HeaderNav.js (hunk 2 above)
Comment (amyrlam) [reply lib/core/nav/HeaderNav.js]: I figured out how to use Prettier in VSCode and am now running `yarn ci-check` to check this before pushing to Circle CI. 📈

Changed to ternary, good suggestion 👍

**Comment by amyrlam** [comment]
> The test plan doesn't seem to match up with the PR and issue. Did you mix up the screenshots?

Hey, I'm not sure what you mean. The screenshots aren't mixed up...but before the `logo` class had a `display:none` on mobile.

I added screenshots for tablet and desktop above in the PR summary.

It would be cool to add snapshot testing down the road, maybe with the task where I add some Jest tests? I'm familiar with Percy.io and it seems like Jest has the same capability.

**Comment by amyrlam** [comment]
That's a good point, I'll do WAS/IS next time.

**Pull request merged**